mne/io/nirx/nirx.py

```diff
@@ -513,3 +512,16 @@ def _convert_fnirs_to_head(trans, fro, to, src_locs, det_locs, ch_locs):
     det_locs = apply_trans(mri_head_t, det_locs)
     ch_locs = apply_trans(mri_head_t, ch_locs)
     return src_locs, det_locs, ch_locs, mri_head_t
+
+
+def _nirs_sort_idx(info):
+    nirs_picks = pick_types(info, fnirs=True, exclude=())
+    other_picks = np.setdiff1d(np.arange(info['nchan']), nirs_picks)
+    prefixes = [info['ch_names'][pick].split()[0] for pick in nirs_picks]
+    nirs_names = [info['ch_names'][pick] for pick in nirs_picks]
+    nirs_sorted = sorted(nirs_names,
```

Comment:
Is there a way we can test that values meet this ordering and place it in... 

https://github.com/mne-tools/mne-python/blob/549de120926465a514ac5d7b0b8950f49ec2ade5/mne/preprocessing/nirs/nirs.py#L87

This would make sure that different processing steps don't modify the ordering. And that the ordering is valid for all files that go through processing such as beer lambert. It is also useful incase users modified the ordering themselves or generated their own data from arrays etc.

Reply:
I'll make a note that we could/should in principle do this. But I'd rather not add it since it will hopefully be almost immediately removed by the next PR to remove any restriction on the ordering